docs/reference/modules/discovery/fault-detection.asciidoc

```diff
@@ -244,141 +248,17 @@ a possible cause for this kind of instability. Log messages containing
-`elasticsearch[follower][cluster_coordination][T#4]`, processes the request. It
-constructs a response and tells the `TransportService` to send the response
-back to the master.
-
-.. The follower's `TransportService`, running on thread
-`elasticsearch[follower][transport_worker][T#3]`, passes the response to the
-operating system.
-
-. The operating system on the follower converts the response into one or more
-packets and sends them out over the network.
-
-. Miscellaneous routers, firewalls, and other devices between master and
-follower forward the packets, possibly fragmenting or defragmenting them on the
-way.
-
-. The operating system on the master receives the packets and notifies {es}
-that they've been received.
-
-. The master's `TransportService`, running on thread
-`elasticsearch[master][transport_worker][T#2]`, reads the incoming packets,
-reconstructs the check response, and processes it as long as the check didn't
-already time out.
-
-There are a lot of different things that can delay the completion of a check
-and cause it to time out. Here are some examples for each step:
-
-. There may be a long garbage collection (GC) or virtual machine (VM) pause
-after passing the check request to the `TransportService`.
-
-. There may be a long wait for the specific `transport_worker` thread to become
-available, or there may be a long GC or VM pause before passing the check
-request onto the operating system.
-
-. A system fault (for example, a broken network card) on the master may delay
-sending the message over the network, possibly indefinitely.
-
-. Intermediate devices may delay, drop, or corrupt packets along the way. The
-operating system for the master will wait and retransmit any unacknowledged or
-corrupted packets up to `net.ipv4.tcp_retries2` times. We recommend
-<<system-config-tcpretries,reducing this value>> since the default represents a
-very long delay.
-
-. A system fault (for example, a broken network card) on the follower may delay
-receiving the message from the network.
-
-. There may be a long wait for the specific `transport_worker` thread to become
-available, or there may be a long GC or VM pause during the processing of the
-request on the follower.
-
-. There may be a long wait for the `cluster_coordination` thread to become
-available, or for the specific `transport_worker` thread to become available
-again. There may also be a long GC or VM pause during the processing of the
-request.
-
-. A system fault (for example, a broken network card) on the follower may delay
-sending the response from the network.
-
-. Intermediate devices may delay, drop, or corrupt packets along the way again,
-causing retransmissions.
-
-. A system fault (for example, a broken network card) on the master may delay
-receiving the message from the network.
-
-. There may be a long wait for the specific `transport_worker` thread to become
-available to process the response, or a long GC or VM pause.
```

Comment:
All of the above has been scrapped completely. Is this intended, and why? Maybe it's not up-to-date, or irrelevant anymore?

Reply:
It's still accurate, but just not that useful for non-developers, and it pushes the actionable info way down the page. I originally added it to answer some questions we received about this, but on reflection I think we could have avoided those questions with the troubleshooting guidance that remains. We can bring it back if we start to see evidence that it was more useful than I now believe it to be.